icu4c/source/i18n/usearch.cpp

```diff
@@ -3626,7 +3656,7 @@ U_NAMESPACE_END
  * has an external break iterator, use that. Otherwise use the internal character
  * break iterator.
  */
-static int32_t nextBoundaryAfter(UStringSearch *strsrch, int32_t startIndex) {
+static int32_t nextBoundaryAfter(UStringSearch *strsrch, int32_t startIndex, UErrorCode *status) {
 #if 0
     const UChar *text = strsrch->search->text;
```

Comment:
In this branch, if U_FAILURE then return immediately; maybe return startIndex?

Reply:
Thanks!